### DISCUSSION TITLE
Foundry crashed while storing PRIVATE_KEY || ERROR - cannot convert a slice of length 20 to FixedBytes<32>

### DISCUSSION THREAD

**Discussion by Nishchit-Dev** [Cyfrin/foundry-full-course-cu#1746] [Q&A]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/83442510/df5ddd87-fa2e-4c08-bbeb-1c7586634350)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/83442510/1b252338-9549-4312-8e36-de7866ee1d8c)

**Getting Error while using foundry inbuilt method to store and encrypt private key**

**Comment by EngrPips** [comment]
Hello @Nishchit-Dev, I tried to store a private key, and it was successful. You can see this in the image below.

<img width="1104" alt="Screenshot 2024-05-23 at 09 44 46" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/21abb15a-8b79-4346-8e5a-168741854159">

I will advise you to run `foundryup` and retry.

**Reply by killer-block-coder-95** [reply]
<img width="692" alt="private key" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/170498606/dc7f894b-30d5-4c08-bbda-231ec37d17b6">

i have stored by private key in PRIVATE_KEY variable. But when I am trying to deploy a contract using the key, I am unable to do so. Please help me out @EngrPips @Nishchit-Dev @thebraveman17 

i tried removing the sollar sign from PRIVATE_KEY, but still get an error
<img width="581" alt="private key 2 " src="https://github.com/Cyfrin/foundry-full-course-f23/assets/170498606/48c33653-273f-4367-aee8-993c7f616813">

**Comment by urosognjenovic** [comment]
Are you sure that it's the private key that you pasted in there and not the address of your account?

I get the same error when pasting an address that has the length of 20 bytes instead of a private key that has the length of 32 bytes.

**Comment by EngrPips** [comment]
Hello @Nishchit-Dev, Please let us know if you have resolved this issue.

**Comment by urosognjenovic** [comment]
You're confusing two different things.

```cast wallet import``` is used to encrypt the private key and store it in the root folder of your computer. After storing it, you can use it to deploy a contract by running ```forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545 --account privateKey --broadcast```. Here, ```privateKey``` is the private key that you imported using ```cast wallet import privateKey```.

If you want to use the environment variables that are stored in your .env file, you can use ```forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545 --private-key $PRIVATE_KEY --broadcast```. The dollar sign is used to instruct the command line to expect the name of an environment variable after the sign. Make sure you've imported the environment variables in the command line by running ```source .env```. To check their availability, run ```echo $PRIVATE_KEY```.

**Comment by EngrPips** [comment]
Hello @Nishchit-Dev, please do as advised by @thebraveman17  and let us know if you successfully resolve the issue.
